pkg/util/grpc.go

```diff
@@ -30,10 +32,16 @@ func CloseAndExhaust[T any](stream Stream[T]) error {
 
 	go func() {
 		for {
-			if _, err := stream.Recv(); err != nil {
+			msg, err := stream.Recv()
+			if err != nil {
 				close(done)
 				return
 			}
+
+			// If the message has a buffer reference, free it.
```

Comment:
nit: Can you add a comment here somewhere that explains that there's nothing to free if the stream returned an error? It took me a second to understand what was going on here.

Reply:
Does what I've added in d2f3a6c make this clearer?